src/assets/wise5/classroomMonitor/student-progress/student-progress.component.ts

```diff
@@ -0,0 +1,238 @@
+      if (team.workgroupId === workgroupId) {
+        team.location = location;
+        team.completion = completion;
+        team.score = score;
+        team.maxScore = maxScore;
+        team.scorePct = maxScore ? score / maxScore : score;
+      }
+    }
+  }
+
+  private getCurrentNodeForWorkgroupId(workgroupId: number): string {
+    return this.classroomStatusService.getCurrentNodePositionAndNodeTitleForWorkgroupId(
+      workgroupId
+    );
+  }
+
+  /**
+   * Get project completion data for the given workgroup (only include nodes with student work)
+   * @param workgroupId the workgroup id
+   * @return object with completed, total, and percent completed (integer between 0 and 100)
+   */
+  private getStudentProjectCompletion(workgroupId: number): any {
+    return this.classroomStatusService.getStudentProjectCompletion(workgroupId, true);
+  }
+
+  private getStudentTotalScore(workgroupId: number): number {
+    return this.teacherDataService.getTotalScoreByWorkgroupId(workgroupId);
+  }
+
+  private sortWorkgroups(): void {
+    this.sortedTeams = [];
+    for (const team of this.teams) {
+      this.sortedTeams.push(team);
+    }
+    switch (this.sort) {
+      case 'team':
+        this.sortedTeams.sort(this.createSortTeam('asc'));
+        break;
+      case '-team':
+        this.sortedTeams.sort(this.createSortTeam('desc'));
+        break;
+      case 'student':
+        this.sortedTeams.sort(this.createSortStudent('asc'));
+        break;
+      case '-student':
+        this.sortedTeams.sort(this.createSortStudent('desc'));
+        break;
+      case 'score':
+        this.sortedTeams.sort(this.createSortScore('asc'));
+        break;
+      case '-score':
+        this.sortedTeams.sort(this.createSortScore('desc'));
+        break;
+      case 'completion':
+        this.sortedTeams.sort(this.createSortCompletion('asc'));
+        break;
+      case '-completion':
+        this.sortedTeams.sort(this.createSortCompletion('desc'));
+        break;
+      case 'location':
+        this.sortedTeams.sort(this.createSortLocation('asc'));
+        break;
+      case '-location':
+        this.sortedTeams.sort(this.createSortLocation('desc'));
+        break;
+    }
+  }
+
+  private createSortTeam(direction: string): any {
+    return (workgroupA: any, workgroupB: any): number => {
+      return direction === 'asc'
+        ? workgroupA.workgroupId - workgroupB.workgroupId
+        : workgroupB.workgroupId - workgroupA.workgroupId;
+    };
+  }
+
+  private createSortStudent(direction: string): any {
+    return (workgroupA: any, workgroupB: any): number => {
+      const localeCompare =
+        direction === 'asc'
+          ? workgroupA.username.localeCompare(workgroupB.username)
+          : workgroupB.username.localeCompare(workgroupA.username);
+      if (localeCompare === 0) {
+        return workgroupA.workgroupId - workgroupB.workgroupId;
+      } else {
+        return localeCompare;
+      }
```

Comment:
This could also be converted to a ternary operator so there's only need for 1 return.